mungegithub/reports/shame.go

```diff
@@ -0,0 +1,280 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reports
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"sort"
+	"strings"
+	"time"
+
+	githubhelper "k8s.io/contrib/mungegithub/github"
+
+	"github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+)
+
+// ShameReport looks for situations where a reviewer has LGTM'd a PR, but it
+// isn't ok to test by the k8s-bot, and adds an 'ok to test' comment to the PR.
+type ShameReport struct {
+	Command string
+	From    string
+	Cc      string
+}
+
+func init() {
+	RegisterReportOrDie(&ShameReport{})
+}
+
+// Name is the name usable in --issue-reports
+func (s *ShameReport) Name() string { return "shame" }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (s *ShameReport) AddFlags(cmd *cobra.Command, config *githubhelper.Config) {
+	cmd.Flags().StringVar(&s.Command, "shame-report-cmd", "tee -a shame.txt", "command to execute, passing the report as stdin")
+	cmd.Flags().StringVar(&s.From, "shame-from", "", "From: header for shame report")
+	cmd.Flags().StringVar(&s.Cc, "shame-cc", "", "Cc: header for shame report")
+}
+
+type reportData struct {
+	loginToEmail     map[string]string
+	loginToIssues    map[string][]issueReportData
+	totalTests       int
+	lowPriorityTests int
+}
+
+type issueReportData struct {
+	number   int
+	age      time.Duration
+	priority string
+	title    string
+}
+
+func (self issueReportData) String() string {
+	days := self.age.Hours() / 24
+	return fmt.Sprintf("  [%v] %v: %q (%.2v days)", self.priority, self.number, self.title, days)
+}
+
+func gatherData(cfg *githubhelper.Config) (*reportData, error) {
+	issues, err := cfg.ListAllIssues(&github.IssueListByRepoOptions{
+		State:  "open",
+		Sort:   "created",
+		Labels: []string{"kind/flake"},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	r := reportData{
+		loginToEmail:  map[string]string{},
+		loginToIssues: map[string][]issueReportData{},
+	}
+	for _, issue := range issues {
+		assignee := "UNASSIGNED"
+		if issue.Assignee != nil && issue.Assignee.Login != nil {
+			assignee = *issue.Assignee.Login
+			if _, ok := r.loginToEmail[assignee]; !ok {
+				if u, err := cfg.GetUser(assignee); err == nil {
+					if u.Email != nil {
+						r.loginToEmail[assignee] = *u.Email
+					} else {
+						// Don't keep looking this up
+						r.loginToEmail[assignee] = ""
+					}
+				}
+			}
+		}
+		age := time.Duration(0)
+		if issue.CreatedAt != nil {
+			age = time.Now().Sub(*issue.CreatedAt)
+		}
+		priority := "??"
+		for _, l := range issue.Labels {
+			if l.Name != nil && strings.HasPrefix(*l.Name, "priority/") {
+				priority = strings.TrimPrefix(*l.Name, "priority/")
+				break
+			}
+		}
+		if priority == "??" {
+			assignee = "UNPRIORITIZED"
```

Comment:
This makes for a nice-ish report, but just because the priority is set doesn't mean that they aren't assigned. If I have an unprioritized flake assigned to me, I should still see it in my list.

Maybe 'unassigned' and 'unprioritized' should be unrelated sections of the report instead of treating them like normal people...